src/game_system.h

```diff
@@ -246,6 +247,9 @@ namespace Game_System {
 
 	void OnBgmReady(FileRequestResult* result);
 	void OnSeReady(FileRequestResult* result, int volume, int tempo, bool stop_sounds);
+
+	bool IsSoundStopFilename(RPG::Sound sound);
+	const RPG::Sound* GetFirstSound(RPG::Animation *animation);
 }
```

Comment:
Imo this whole commit is overcomplecating the problem.
The only thing you want to do is: Play the first SE of the animation, otherwise nothing.

So why don't you add a new function void ``SePlay(const RPG::Animation& animation)`` which simply delegates the work to the other existing SePlay. Then you can also remove IsSoundStopFilename

```
/**
 * Plays the first sound effect of an animation if it has one.
```

Reply:
This doesn't solve the problem, which is "if it's a sound that is not (OFF) or similars, stop looking at more sound effect animations", and there's no way to know by executing SePlay if the sound was (OFF) or not. Remember that every effect executes the sound "(OFF)". That's why I need some kind of check. And I didn't want to use the default IsStopFilename because that searches for the files and all that, and that could be not efficient, specially in a function that is looping until it find a valid sound.

Reply:
The filenames are cached so calling the FileFinder hundreds of times is fine but I didn't consider how finding the first SE works, so have to rethink this.

Reply:
well I would prefer a SePlay(const RPG::Animation& animation) which uses the normal "IsStopFilename" func that already exists and then delegates the work to SePlay when it's not a StopFilename.
This will fail for a small corner case in emscripten (SE start/ends with ( / ) ) but I don't see a good way to solve this for this.